Makefile.e2e

```diff
@@ -59,17 +59,34 @@ test-e2e-gke-nobuild:
 .PHONY: test-e2e-gke
 test-e2e-gke: config-sync-manifest test-e2e-gke-nobuild
 
+KIND_IMAGE := "kind-image"
+
+.PHONY: build-kind-image
+build-e2e-kind: config-sync-manifest-local build-gke-e2e
```

Comment:
`build-e2e-kind` doesn't need to depend on `config-sync-manifest-local`. We can make `config-sync-manifest-local` a dependency of `test-e2e-kind-multi-repo`.

Reply:
Done. To accomplish this I did some refactoring to allow running `config-sync-manifest-local` without a recursive make call. This preserves the dependency tree so certain targets don't get run multiple times (e.g. `install-helm`)